File: rewrite_theme.js
const fs = require('fs');

const path = 'd:/Rnd files/event-analytics-main/src/features/analytics/pages/KpiDashboard.tsx';
let content = fs.readFileSync(path, 'utf8');

const replacements = [
    [/bg-\[\#0B0F19\]/g, 'bg-slate-50 dark:bg-[#0B0F19]'],
    [/bg-\[\#111827\]\/50/g, 'bg-white dark:bg-[#111827]/50'],
    [/bg-\[\#111827\](?!\/)/g, 'bg-white dark:bg-[#111827]'],
    [/bg-\[\#0F172A\]/g, 'bg-slate-50 dark:bg-[#0F172A]'],
    [/bg-white\/\[0\.02\]/g, 'bg-slate-100 dark:bg-white/[0.02]'],
    [/border-white\/5(?!0)/g, 'border-slate-200 dark:border-white/5'],
    [/border-white\/10/g, 'border-slate-300 dark:border-white/10'],
    [/border-white\/20/g, 'border-slate-300 dark:border-white/20'],
    [/border-\[\#1E293B\]/g, 'border-slate-200 dark:border-[#1E293B]'],
    [/(?<!dark:)text-white(?![A-Za-z0-9\-\/])/g, 'text-slate-900 dark:text-white'],
    [/(?<!dark:)text-slate-200(?![A-Za-z0-9\-\/])/g, 'text-slate-700 dark:text-slate-200'],
    [/(?<!dark:)text-slate-300(?![A-Za-z0-9\-\/])/g, 'text-slate-600 dark:text-slate-300'],
    [/(?<!dark:)text-slate-400(?![A-Za-z0-9\-\/])/g, 'text-slate-500 dark:text-slate-400']
];

for (const [pattern, replacement] of replacements) {
    content = content.replace(pattern, replacement);
}

// Special fixes for node properties which take an explicit stroke/fill which don't support tailwind dark mode:
// Recharts doesn't natively support classNames nicely for all SVG colors without explicitly passing CSS variables.
// A simpler robust way? Leave Recharts node stroke as '#0F172A'. Wait, what if the chart is on white bg? 
// The stroke is `#0F172A` (dark blue). On a light background, a dark border is actually perfect!
// The tooltip background can be dynamic but for now let's leave it as is.
// Let's write the content back.
fs.writeFileSync(path, content, 'utf8');
console.log('Saved properly!');

File: rewrite_theme.py
import re

path = r"d:\Rnd files\event-analytics-main\src\features\analytics\pages\KpiDashboard.tsx"
with open(path, "r", encoding="utf-8") as f:
    content = f.read()

replacements = [
    (r"bg-\[\#0B0F19\]", "bg-slate-50 dark:bg-[#0B0F19]"),
    (r"bg-\[\#111827\]/50", "bg-white dark:bg-[#111827]/50"),
    (r"bg-\[\#111827\](?!\/)", "bg-white dark:bg-[#111827]"),
    (r"bg-\[\#0F172A\]", "bg-slate-50 dark:bg-[#0F172A]"),
    (r"bg-white/\[0\.02\]", "bg-slate-100/50 dark:bg-white/[0.02]"),
    (r"border-white/5(?!0)", "border-slate-200 dark:border-white/5"),
    (r"border-white/10", "border-slate-300 dark:border-white/10"),
    (r"border-white/20", "border-slate-300 dark:border-white/20"),
    (r"border-\[\#1E293B\]", "border-slate-200 dark:border-[#1E293B]"),
    (r"(?<!dark:)text-white(?![A-Za-z0-9\-\/])", "text-slate-900 dark:text-white"),
    (r"(?<!dark:)text-slate-200(?![A-Za-z0-9\-\/])", "text-slate-700 dark:text-slate-200"),
    (r"(?<!dark:)text-slate-300(?![A-Za-z0-9\-\/])", "text-slate-600 dark:text-slate-300"),
    (r"(?<!dark:)text-slate-400(?![A-Za-z0-9\-\/])", "text-slate-500 dark:text-slate-400"),
    (r"(?<!dark:)text-slate-500(?![A-Za-z0-9\-\/])", "text-slate-400 dark:text-slate-500"),
]

for pattern, replacement in replacements:
    content = re.sub(pattern, replacement, content)

with open(path, "w", encoding="utf-8") as f:
    f.write(content)

print('Success')
